proposals/NativeInt.md

```diff
@@ -0,0 +1,263 @@
+
+_Are these "identity" conversions between equivalent types represented in the Roslyn API?_
+
+The following numeric conversions are supported.
+(The IL for each conversion includes the variants for `unchecked` and `checked` contexts if different.)
+
+| Operand | Target | Implicit | IL |
+|:---:|:---:|:---:|:---:|
+| `sbyte` | `nint` | Y | `conv.i` |
+| `byte` | `nint` | Y | `conv.i` |
+| `short` | `nint` | Y | `conv.i` |
+| `ushort` | `nint` | Y | `conv.i` |
+| `int` | `nint` | Y | `conv.i` |
+| `uint` | `nint` |   | `conv.i` / `conv.ovf.i` |
+| `long` | `nint` |   | `conv.i` / `conv.ovf.i` |
+| `ulong` | `nint` |   | `conv.i` / `conv.ovf.i` |
+| `char` | `nint` | Y | `conv.i` |
+| `float` | `nint` |   | `conv.i` / `conv.ovf.i` |
+| `double` | `nint` |   | `conv.i` / `conv.ovf.i` |
+| `sbyte` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `byte` | `nuint` | Y | `conv.u` |
+| `short` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `ushort` | `nuint` | Y | `conv.u` |
+| `int` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `uint` | `nuint` | Y | `conv.u` |
+| `long` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `ulong` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `char` | `nuint` | Y | `conv.u` |
+| `float` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `double` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+
+| Operand | Target | Implicit | IL |
+|:---:|:---:|:---:|:---:|
+| `nint` | `nuint` |   | `conv.u` / `conv.ovf.u` |
+| `nint` | `sbyte` |   | `conv.i1` / `conv.ovf.i1` |
+| `nint` | `byte` |   | `conv.u1` / `conv.ovf.u1` |
+| `nint` | `short` |   | `conv.i2` / `conv.ovf.i2` |
+| `nint` | `ushort` |   | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `int` |   | `conv.i4` / `conv.ovf.i4` |
+| `nint` | `uint` |   | `conv.u4` / `conv.ovf.u4` |
+| `nint` | `long` | Y | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `ulong` |   | `conv.u8` / `conv.ovf.u8` |
+| `nint` | `char` |   | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `float` | Y | `conv.r4` |
+| `nint` | `double` | Y | `conv.r8` |
+| `nuint` | `nint` |   | `conv.i` / `conv.ovf.i` |
+| `nuint` | `sbyte` |   | `conv.i1` / `conv.ovf.i1` |
+| `nuint` | `byte` |   | `conv.u1` / `conv.ovf.u1` |
+| `nuint` | `short` |   | `conv.i2` / `conv.ovf.i2` |
+| `nuint` | `ushort` |   | `conv.u2` / `conv.ovf.u2` |
+| `nuint` | `int` |   | `conv.i4` / `conv.ovf.i4` |
+| `nuint` | `uint` |   | `conv.u4` / `conv.ovf.u4` |
+| `nuint` | `long` |   | `conv.i8` / `conv.ovf.i8` |
+| `nuint` | `ulong` | Y | `conv.u8` / `conv.ovf.u8` |
+| `nuint` | `char` |   | `conv.u2` / `conv.ovf.u2.un` |
+| `nuint` | `float` | Y | `conv.r.un conv.r4` |
+| `nuint` | `double` | Y | `conv.r.un conv.r8` |
+
+Conversions between `decimal` and `nint` or `nuint` are not supported because there are no operators on `decimal` that use `native int`.
+There are operators on `decimal` to convert to and from `long` or `ulong` but those are not optimal on 32-bit platforms.
+
+### Operators
+
+The following operators are supported.
```

Comment:
>are supported [](start = 24, length = 13)

Do you mean to say that these are built-in operators considered when seeking to determine the semantics of an operation in source, like the built-in ones described in [https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#addition-operator](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#addition-operator) ?